probe-rs/src/debug/unit_info.rs

```diff
@@ -1355,6 +1303,139 @@ impl UnitInfo {
         Ok(())
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn extract_enumeration_type(
+        &self,
+        child_variable: &mut Variable,
+        type_name: Option<String>,
+        debug_info: &DebugInfo,
+        node: &DebuggingInformationEntry<GimliReader>,
+        parent_variable: &Variable,
+        memory: &mut dyn MemoryInterface,
+        frame_info: StackFrameInfo,
+    ) -> Result<(), DebugError> {
+        child_variable.type_name =
+            VariableType::Enum(type_name.unwrap_or_else(|| "<unnamed enum>".to_string()));
+
+        self.process_memory_location(
+            debug_info,
+            node,
+            parent_variable,
+            child_variable,
+            memory,
+            frame_info,
+        )?;
+
+        let mut tree = self.unit.entries_tree(Some(node.offset()))?;
+        let enumerator_values = self.process_enumerator(debug_info, tree.root()?)?;
+
+        if !(parent_variable.is_valid() && child_variable.is_valid()) {
+            return Ok(());
+        }
+
+        let value = if let VariableLocation::Address(address) = child_variable.memory_location {
+            // NOTE: hard-coding value of variable.byte_size to 1 ... replace with code if necessary.
+            let mut buff = 0u8;
+            memory.read(address, std::slice::from_mut(&mut buff))?;
+            let this_enum_const_value = buff.to_string();
+
+            let enumumerator_value = match enumerator_values
+                .iter()
+                .find(|(_name, value)| value.to_string() == this_enum_const_value)
+            {
+                Some((name, _value)) => name,
+                None => &VariableName::Named("<Error: Unresolved enum value>".to_string()),
+            };
+
+            self.language
+                .format_enum_value(&child_variable.type_name, enumumerator_value)
+        } else {
+            VariableValue::Error(format!(
+                "Unsupported variable location {:?}",
+                child_variable.memory_location
+            ))
+        };
+
+        child_variable.set_value(value);
+
+        Ok(())
+    }
+
+    /// Extract the different variants of an enumeration
+    ///
+    /// This is used for C-style enums, where the enum is an integer type,
+    /// and all the different variants are different integer values.
+    fn process_enumerator(
+        &self,
+        debug_info: &DebugInfo,
+        parent_node: gimli::EntriesTreeNode<GimliReader>,
+    ) -> Result<Vec<(VariableName, VariableValue)>, DebugError> {
+        let mut enumerator_values = Vec::new();
+
+        let mut child_nodes = parent_node.children();
+        while let Some(child_node) = child_nodes.next()? {
+            match child_node.entry().tag() {
+                gimli::DW_TAG_enumerator => {
+                    let attributes_entry = child_node.entry();
```

Comment:
It's just a style thing, but since we only handle `DW_TAG_enumerator` in this loop, would you `continue` in the other match arms, and pull this whole block out to after the match, please?

Reply:
The loop is pointless now, I agree. I'll just replace it with getting the attribute directly, because we only care about a single one.